test/functional/feature_bip8.py

```diff
@@ -0,0 +1,124 @@
+#!/usr/bin/env python3
+# Copyright (c) 2021 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+# Test BIP 8 softforks
+from collections import namedtuple
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.util import assert_equal
+
+RestartedNodeInfo = namedtuple("RestartedNodeInfo", ("node", "extra_args", "status"))
+
+class Bip8Test(BitcoinTestFramework):
+    def set_test_params(self):
+        self.num_nodes = 4
+        self.setup_clean_chain = True
+        self.extra_args = [
+            ['-vbparams=testdummy:@-2:@-2'], # Node 0 has TestDummy inactive
+            ['-vbparams=testdummy:@144:@{}'.format(144 * 3)], # Node 1 has regular activation window
```

Comment:
f-strings are generally preferred in new tests:

```suggestion
            [f'-vbparams=testdummy:@144:@{144 * 3}'], # Node 1 has regular activation window
```

Reply:
These use this format so that it is easier to backport. 0.21 is using a version python that doesn't support f-strings.